support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/validator/authorization/OAuth20MissingOrInvalidResponseTypeAuthorizationRequestValidator.java

```diff
@@ -0,0 +1,35 @@
+package org.apereo.cas.support.oauth.validator.authorization;
+
+import org.apereo.cas.audit.AuditableExecution;
+import org.apereo.cas.authentication.principal.ServiceFactory;
+import org.apereo.cas.authentication.principal.WebApplicationService;
+import org.apereo.cas.services.ServicesManager;
+import org.apereo.cas.support.oauth.OAuth20Constants;
+import org.apereo.cas.support.oauth.OAuth20ResponseTypes;
+import org.apereo.cas.support.oauth.util.OAuth20Utils;
+
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.JEEContext;
+
+/**
+ * This is {@link OAuth20MissingOrInvalidResponseTypeAuthorizationRequestValidator}.
+ *
+ * @author Julien Huon
+ * @since 6.3.0
+ */
+public class OAuth20MissingOrInvalidResponseTypeAuthorizationRequestValidator extends OAuth20AuthorizationCodeResponseTypeAuthorizationRequestValidator {
```

Comment:
The class name seems rather odd to me. How about `OAuth20ResponseTypeAuthorizationRequestValidator` instead? You're validating the response type, and the validator contains a number of checks. Those checks don't need to be advertised in the class name.

Reply:
I wanted to stay consistent with the other class names but you're proposal is better. Will do it. :)

Reply:
Just a question.

All the current validators are related to the response_type parameter and if an authorization request is made without any response_type, none of the validators matches and it is rejected.

With this PR I want to still refuse them but to have all the error details in the context. Should I just rename the class or try to find another supports logic in the Authorization validators?

Reply:
I see. In that case, none of the above :) 

What you should do is: in the super parent class, implement the response_type checking logic that you have here. When that passes, then you call an abstract protected method (defined by the super parent class) that is implemented by other validators who can do the job. This way, the common logic is contained within the parent class and not duplicated everywhere. It also helps you remove this class.

See `AbstractPreAndPostProcessingAuthenticationHandler#authenticate` for an example of doing pre/execution/post.

Reply:
Still have a question, sorry.

Today, the right class is selected via a stream which is looking to a class where the supports method will respond true. All theses supports methods are looking for a particular value in the response_type parameter. When the right class is found, the validate method of the class found is called.

```
val validator = oauthAuthorizationRequestValidators
            .stream()
            .filter(b -> b.supports(context))
            .findFirst()
            .orElse(null);

        if (validator == null) {
            context.setRequestAttribute(OAuth20Constants.ERROR, OAuth20Constants.INVALID_REQUEST);
            return false;
        }
        return validator.validate(context);
```

So, I guess I'll have to pre-validate the request in the supports method if we want to remove this class. Because, even if I do what you want, the validate method will never be called.

I could add the logic needed and the error details will be in the context but an invalid request gonna be checked multiple times (as much as the number of validate class defined in the oauthAuthorizationRequestValidators class) instead of being rejected instantly. 

Because a request without or with an invalid request could produce a callback if the client_id & the redirect_uri do exist in the request. I need to check them in pre-validate method which is more expensive than looking to a single parameter.

 In the code, it's more elegant but I don't known if it's more performant. Unless I missed something?

Reply:
> So, I guess I'll have to pre-validate the request in the supports method if we want to remove this class. 

That makes sense. You'd apply the strategy I pointed out earlier to the `supports` method. pre-validate in a parent class, and if it passes, let each child authorizer class decide if it can support the request. There should be no performance side-effects with multiple checking of the parameter's existence or value. My main concern is maintainability and readability. Let's make it work, and make it look good. Then, we'll make it fast[er] :) 


>Because a request without or with an invalid request could produce a callback if the client_id & the redirect_uri do exist in the request. I need to check them in pre-validate method which is more expensive than looking to a single parameter.

Can you elaborate more please? By callback, you mean a redirect back to the application if the client_id & the redirect_uri do exist?

Reply:
> Can you elaborate more please? By callback, you mean a redirect back to the application if the client_id & the redirect_uri do exist?

Yes. When the client_id & the redirect_uri is valid you have to redirect the user-agent to the redirect_uri with parameters : state if any, error and an error_description if any.

So a big part of the actual validate method will be in the pre-validate method : check if a client_id & a redirect_uri do exist. Does the client_id exists in our configuration? Does the redirect_uri is valid for the client_id.

Reply:
That makes sense to me. If pre-validate passes, you can then invoke validate, or optionally issue a redirect if data is available.

Reply:
Ok I'll do it.

About the coverage, the code produced has a slightly better coverage than the existent one. If I can do even more I will. :)

Reply:
Done.